src/pages/index.js

```diff
@@ -2,76 +2,215 @@ import "./index.css";
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const itemSection = new Section({
-  items: initialCards,
-  renderer: renderCard,
-  cssSelector: ".cards__list",
+let itemSection;
+
+api.getCardList().then((items) => {
+  itemSection = new Section({
+    items,
+    renderer: renderCard,
+    cssSelector: ".cards__list",
+  });
+
+  itemSection.renderItems(items);
 });
-itemSection.renderItems();
 
-const addCardModal = new PopupWithForm(
-  "#profile-add-card",
-  handleCardAddSubmit
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
 
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    // a = true
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+    deleteConfirm.close();
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
+function handleLikeClick(card) {
+  api
+    .toggleCardLike(card._id, card.isLiked)
+    .then((_isLiked) => {
+      // console.log("RESPONSE");
+      // console.log(_isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleChangePicture(avatar) {
+  changePictureModal.setEventListeners();
+  console.log(avatar);
+  this._buttonElement = this._popupForm.querySelector(".modal__button");
+  this._buttonElement;
+  changePictureModal.renderLoading(true);
+  api
+    .setUserAvatar(avatar)
+    .then((data) => {
+      changePictureModal.renderLoading(false);
+      console.log(data.avatar);
+      userInfo.setUserAvatar(data.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  changePictureModal.close();
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  handleChangePicture
+);
+changePictureModal.setEventListeners();
+const pictureFormValidator = new FormValidator(config, pictureChangeForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```